### DISCUSSION TITLE
[Balance] Crafting recipes and resources (post 3y update)

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#9399] [Unstable feedback]
Overall the "scarcity" update was pretty good, for the first time it felt like the sub could actually run out of resources.

Infinite deconstruction loops are gone, "recycle" recipes are finally useful but most items have seem their resource cost increase 2 fold.
This isn't so much a problem for cheap items (like wrenches), but the balance was thrown off, mostly for consumables like ammunition.

Based on what resources we were sorely missing, and resources we had a massive surplus of, here's a list:

**Common surplus:**
Iron, Silicon, Tin, Calcium, Phosphorous, Aluminium, Organic fiber, Uranium*

**Very scarce resources:**
Steel, Lithium, Lead, Magnesium*

### List of recipes that need change

**Steel Bar**
The most consistently used/requested material, it is needed for everything, more than ever before. You get plenty of Iron, but not enough Carbon. From experience we found ourselves consistently starved for steel
_Proposed change:_  Iron + Carbon(x1), reduce deconstructor yield to 1x Carbon

**Railgun shell** 
2x Steel per shot made the weapon too expensive to shoot
_Proposed change:_ Steel + Iron would be more balanced

**Physicorium Shell**
2x Physicorum can be very expensive; or not (see talents at the bottom)
_Proposed change:_ Physicorium + Iron (or steel)

**Harpoon:**
Doubled the cost for the same amount
_Proposed change:_ x6 harpoons instead of 3x

**Physicorium Harpoon**
Same
_Proposed change:_ x6 harpoons instead of 3x

**Physicorium Coilgun ammo**
Doubled the cost for the same amount.
_Proposed change:_ 1x Physicorium + 1 Steel + 1 Aluminum, reduce deconstructor  yield to 1x Physicorium

**Physicorium Chaingun ammo**
Same
_Proposed change:_ 1x Physicorium + 1 Titanium + 1 Aluminum, reduce deconstructor yield to 1x Physicorium

**Chaingun ammo**
Doubled the Lead cost for the same amount; Lead is used a lot now
_Proposed change:_ 1x Lead + 1 Titanium + 1 Aluminum, reduce deconstructor yield to 1x Lead

**Revolver rounds:**
Not enough yield for the cost and not very powerful. 1 Magnesium that could be used for anything else.
_Proposed change:_ x12 bullets instead of x4

**Depleted fuel revolver rounds:**
Not enough yield for the cost and not very powerful. 1 Magnesium that could be used for anything else.
_Proposed change:_ x8 bullets instead of x6

**Repair pack:**
Copper + Silicon+ Steel. A consumable that uses half a steel bar to repair slightly faster is a no-no + the talent that unlocks it already competes with the very useful fabricator tinkering, so I doubt anyone will pick that.
_Proposed change:_ Tin + Silicon + Iron (all 3 surplus)

### Other economic fuckery:

**Surplus materials**
-Silicon/Tin don't have many uses so you end up stockpiling them.
-Phosphorous/Calcium you get from looting Mudraptors
-Iron because you find wrenches/screwdrivers everywhere, and iron isn't used a lot

**Starter Gear**
-Respawning mechanics is like hitting the lottery: free plasma cutter and welding tools, which both deconstruct into precious steel, plastic, and have aluminum cans inside them, and a toolbelt for good measure (more fiber)
-Organic fiber is now unlimited as players respawn since clothes deconstruct into it.
-Copper from headsets, for the same reason as above. _Headsets need a yield nerf_

**Talents**
-[Arms Dealer](https://barotraumagame.com/wiki/Arms_Dealer) is the main source of magnesium in the game. It's pretty busted, as you can get it pretty early and will get you an insane amount of free, passive resources.
-[Nuclear option](https://barotraumagame.com/wiki/Nuclear_Option) completely breaks fuel as a mechanic. Getting 3x high quality rods, you basically need three times less lead and produce 3x as much depleted fuel on command. *Remove the 3x rod aspect, and increase rod durability in fissile accelerator*.
-[Xenologist branch for medics](https://barotraumagame.com/wiki/Talents#Medical_Doctor_Talents) will give you absurd amount of alien materials (and/or) money from looting ruins. A single ruins run can get you 10+ alien trinkets which deconstruct into 15 materials on average, plus all of the raw materials found. Look for [Scounger/Scrap savant](https://barotraumagame.com/wiki/Talents#Mechanic_Talents) for a more balanced version of this
-[Miner](https://barotraumagame.com/wiki/Talents#Mechanic_Talents) giving double the output is a 50% increase of mining yields, this is pretty huge for a T2, and the game will send you into mining missions regularly. See below

**Mining missions/ore**
These are the guaranteed rewards of all the minimum missions, *excluding other materials you can find*:
-Quartz: (20 silicon)
-Pyromorphite: (20 chlorine)
-Titanite: 18 Titanium
-Thorianite: 12 Thorium
-Uranium: 24 Uranium 
Silicone and Chlorine aren't really useful, Titanite is near unlimited chaingun ammunition (although bottlednecked by Lead), but most importantly a single Thorium/Uranium mission gives fuel for the entire campaign. 
Yield probably needs to be nerfed for both of these.

**Comment by Oukido** [comment]
>Common surplus:
>Iron, Silicon, Tin, Calcium, Phosphorous, Organic fiber, Lead, Uranium*
>
>Very scarce resources:
>Steel, Lithium, Magnesium*

Carbon was never a problem for me since you could grow carbon with plants and deconstruct them. So as long as I had surplus Iron I also had abundant steel. This carbon source is commonly overlooked.

Lithium i got alot of by deconstructing batteries in the past which is way more scarce now with the headset change. For me it is basically a disqualifier for lasers now.

**Comment by JeroonBosch** [comment]
> Overall the "scarcity" update was pretty good, for the first time it felt like the sub could actually run out of resources.

Glad you see this from the same perspective as I did. In my own tests I got the same feeling, which was the main intention behind these changes.

> Carbon was never a problem for me since you could grow carbon with plants and deconstruct them. So as long as I had surplus Iron I also had abundant steel. This carbon source is commonly overlooked.

This was my thought as well: Carbon is a renewable, near infinite resource. Hence why I chose to increase the Steel recipe with +1 carbon. It might be too bothersome to keep deconstructing plants just to get carbon for steel though, so I'll definitely look into this still.

Aside from that, I am aware these changes are by no means perfect, so this feedback is very welcome. There's a lot of aspects affecting balance, thus getting it right will be an iterative and gradual process. Feedback like this is therefore very welcome. Some of the things mentioned were already on our radar, but we simply haven't gotten around to yet:

- Handheld weapon ammo used to be way too expensive (f.e. 200 coilgun shots having comparable costs to 6 revolver bullets made no sense, nor did 1 harpoon = 1 railgun shell in costs). A small pass over this has been done, but more drastic changes are needed there still
- Materials used to be very easy to come by due to how abundant wreck loot was. This has been reduced a lot now, which might have caused too much scarcity. There's of course many approaches to fix this, of which probably mix would be best: 
1. Less recipes using Steel (and instead using Iron)
2. Upping the loot in f.e. beacon stations and wrecks (especially the items that deconstructs to steel)
3. Easier access to Carbon (aside from grown plants)
- Mining missions and cave/abyss resources will be seeing some adjustments for sure, which should also alleviate some issues
- Talents are definitely quite imbalanced in some cases, so we'll have to look into these as well for sure

Final thoughts: approaching balance from a point of scarcity is much easier than from a point of abundance, as it's much easier to collect feedback on what's missing. This might mean the game is too hard in this particular update, but should hopefully make the game better overall in the long term.

**Reply by ubertpendragon** [reply]
Other observations after 2 days of campaigning, I revised some of my Surplus/Scarce list

**Recycle recipes don't actually work most of the time.** 
I was wrong about recycling (mostly)
Take for example Coilgun Ammo:

Crafting: 2x Lead, 1x Steel, 1x Aluminium
Recycle: 2x Lead, 1x Steel, 1x Empty Ammo Box
Deconstruction for empty ammo box: 1x Aluminium

**Recycle recipes as a whole should always use 1 less resource than their crafting counterpart**; otherwise it allows you to run things through the deconstruction loop for better quality (and skills).
For example, the Tri-laser recycle recipe is only 1x Fulgurium, while making a box from scratch costs 2x. Same for Piercing coilgun ammo or railgun canister shells which are cheaper.

Since ammo boxes are typically 2x Material1, 1x Material2 and 1x Aluminum, the recipe should be either:
-1x Material1 1x Material2 1x Empty box
-2x Material1 1x Empty box

Other "perfect deconstruction" items are:
-Welding fuel tanks; deconstruct into Aluminium when used
-SMG magazines (deconstruct into steel)

Aluminium surplus is pretty high too; it wouldnt hurt to increase the cost of these recipes, for example:
-Welding fuel tank: 1x Al 1x Ethanol => 2x Al 1 Ethanol


**Stations and stealing**

It's a bit of an issue when every single crew seems to steal from outposts without repercussions. Some rooms like this one in Engineeringmodule_02 gives a free suit and 3x oxygen tanks every time.
![image](https://user-images.githubusercontent.com/73229309/175177719-4acd1285-16b9-4808-af68-7e75ac509bac.png)



> _Carbon was never a problem for me since you could grow carbon with plants and deconstruct them. So as long as I had surplus Iron I also had abundant steel. This carbon source is commonly overlooked._

> _This was my thought as well: Carbon is a renewable, near infinite resource. Hence why I chose to increase the Steel recipe with +1 carbon. It might be too bothersome to keep deconstructing plants just to get carbon for steel though, so I'll definitely look into this still._

Carbon is technically infinite since you can spam tobacco. Plants have no limit on how many you can have, spamming planter pots prints free money even after the nerfs (which also made plants more resilient to floods). 

The balance is very different for those who partake in industrial plant farming and those who don't

**Reply by Krzeszny** [reply]
One more perfect deconstruction: **Handcannon**
> The balance is very different for those who partake in industrial plant farming and those who don't

This is why I think it would be better to make seeds deconstruct into nothing, but craft steel and plastic from 1x carbon

**Comment by Oukido** [comment]
A thought about uranium:
Often times late game players find themselves in a situation where they have stacks of uranium. But we have to keep in mind that most experienced players will go for the 75% fuel efficiency upgrade asap, which reduces the need for fuel immensly ontop of other efficiency upgrades which reduce power draw, which indirectly also affects fuel usage.
Beginner players do not have access to those early on and are dependent on regular fuel income. Especially if they are still figuring things out and take way longer for their missions.
Before reducing the availability of uranium (or fuel sources in general) it might be better to reduce the effect of the fuel efficiency as a way to increase the demand on fuel sources.

**Reply by Krzeszny** [reply]
With scarcity or without, reactor control circuits could be rebalanced. You can underclock fission to 75% and undervolt output to as low as 50%, reducing fuel usage by 25%-62%. 

Right now reactor computers are only useful for faster recovery anyway. That's because fuel rods are limitless, _unless_ you use the fission accelerator. Then it's free ammo, but I haven't seen anyone using it for weeks.

Also note that the Fissile Accelerator has:
4 shots from Uranium rods
4 shots from Thorium rods
8 shots from Fulgurium rods
**5** shots from Volatile Fulgurium rods - why would *anyone* waste fulgurium and thorium to make it?

**Reply by JeroonBosch** [reply]
> Also note that the Fissile Accelerator has:
> 4 shots from Uranium rods
> 4 shots from Thorium rods
> 8 shots from Fulgurium rods
> 5 shots from Volatile Fulgurium rods - why would anyone waste fulgurium and thorium to make it?

This is not correct, Volatile Fulgurium rods take 20 shots, while Fulgurium rods take 12.
Volatile Fulgurium rods have 400 health and lose 20 per shot (400 / 20 = 20)
Fulgurium rods have 150 health and lose 12.5 per shot (=12)

**Reply by Krzeszny** [reply]
I always keep forgetting about health which isn't typical to items.
Thorium fuel rods have 200 health.

Correct version:
4 shots from Uranium rods
8 shots from Thorium rods
12 shots from Fulgurium rods
20 shots from VF rods

VF rods save 2 lead compared to using 1 thorium rod and 1 fulgurium rod. Not amazing but 20 shots instead of 8 and 12 is okay.

**Comment by Ek-Videogames** [comment]
Making deconstructor violate equivilent-exchange seemed kind of lazy & hamfisted. Shop price changes seemed like they were more in the right direction - only problem is they weren't actually "fixed" yet just made 2x as expensive to buy and 1/4 as profitable to sell. So while that has removed from the player any possibility of 'trade profiteering', the underlying price inconsistencies are still a thing to address in the future.

I still find it quite odd that apparently the deconstructor change wasn't even in response to economy to begin with - it was something about XP grinding nonsense. Cannibalizing equipment or recycling unusable items just seemed like a natural part of gameplay & allocation of resources.

> -Silicon/Tin don't have many uses so you end up stockpiling them.
-Phosphorous/Calcium you get from looting Mudraptors

Only problem with making these 'specialty' resources even scarcer / making them used more commonly by recipes, is it then is way easier to get screwed because of RNG and be unable to obtain any to craft a certain critical recipe. Also raptors exist for like 8% of the game.
_(It'd be nice to see large monsters in the later biomes actually drop useable loot especially since you're beyond the outposts to buy anything at that point in campaign.)_

- As I said in the unstable discussions, mining kind of sucks, is destroys the pace of the game plus multiplayer servers get really unstable when rounds drag on too long so that's why nobody usually does it. Also you are at mercy of level-generation as to weather or not you find the minerals you're looking for _(aluminum, titanium, & obscure chemical components being the most annoying.)_ and there was better ways to get resources usually.

### **Edit:** 
-and as usual, the fallacy of "assuming all balance changes in a vaccum."
It feels like the intended issue could have been solved by doing one or two of the three major balance changes. Item-scarcity, crafting-cost, and `(actually nevermind F the deconstructor change completely)`, but instead of picking one doing all three at once has caused some overlap of their compounding effects that - as a whole - ends up swinging way too far the other way. A uncommon good-example given by uber being that everything is so scarce now and because of the increased crafting costs demand is even higher than before.

> "Handheld weapon ammo used to be way too expensive (f.e. 200 coilgun shots having comparable costs to 6 revolver bullets made no sense, nor did 1 harpoon = 1 railgun shell in costs). A small pass over this has been done, but more drastic changes are needed there still"

I was under the impression that was intentional, given how absolutely vital ship ammunition is. While small-arms ammo `(for a very long time not craftable to begin with)` were more of a specialty last line of defense & not to be commonly used for dealing with ship threats so could afford to be more expensive.
